actors/miner/src/lib.rs

```diff
@@ -3273,36 +3286,188 @@ impl Actor {
                 // Verify unlocked funds cover both InitialPledgeRequirement and FeeDebt
                 // and repay fee debt now.
                 let fee_to_burn = repay_debts_or_abort(rt, state)?;
-
-                Ok((info, newly_vested, fee_to_burn, available_balance, state.clone()))
+                let mut amount_withdrawn = std::cmp::min(&available_balance, &params.amount_requested);
+                if amount_withdrawn.is_negative() {
+                        return Err(actor_error!(
+                            illegal_state,
+                            "negative amount to withdraw: {}",
+                            amount_withdrawn
+                        ));
+                }
+                if info.beneficiary != info.owner {
+                    let remaining_quota = info.beneficiary_term.available(rt.curr_epoch());
+                    if remaining_quota.is_negative() {
+                        return Err(actor_error!(
+                            forbidden,
+                            "quota not enough beneficiary {}  quota {} used quota {} expiration epoch {}  current epoch {}",
```

Comment:
Hmm, this is quite a bad state to get into. I wonder if we should treat it more gracefully?
Is there any way for the Miner to recover from this state?

Reply:
Looking at it, `available` will never return a negative quota, so this can be removed.

Reply:
you are right

Reply:
done